src/fprime/fbuild/interaction.py

```diff
@@ -216,3 +270,144 @@ def new_component(
     except OSError as ose:
         print("[ERROR] {}".format(ose))
     return 1
+
+def get_port_input():
+    # Gather inputs to use as context for the port template
+    defaults = {
+        "port_name": "ExamplePort",
+        "short_description" : "Example usage of port",
+        "dir_name" : "example_directory",
+        "arg_number" : 1,
+    }
+    valid_name = False
+    valid_dir  = False
+    invalid_characters = ["#", "%", "&", "{", "}", "/", "\\", "<", ">", "*", "?",
+                        " ", "$", "!", "\'", "\"", ":", "@", "+", "`", "|", "="]
+    while not valid_name:
+        port_name = input("Port Name [{}]: ".format(defaults["port_name"]))
+        valid_name = True
+        for char in invalid_characters:
+            if char in port_name:
+                valid_name = False
+                print("'" + char + "' is not a valid character. Enter a new name:")
+    short_description = input("Short Description [{}]: ".format(defaults["short_description"]))
+    while not valid_dir:
+        dir_name = input("Directory Name [{}]: ".format(defaults["dir_name"]))
+        valid_dir = True
+        for char in invalid_characters:
+            if char in dir_name:
+                valid_name = False
+                print("'" + char + "' is not a valid character. Enter a new directory name:")
+    string_arg_number = input("Number of arguments [{}]: ".format(defaults["arg_number"]))
+    if string_arg_number == "":
+            arg_number = 1
+    elif not string_arg_number.isnumeric():
+        print("[ERROR] You have not entered a valid number")
+    else:
+        arg_number = int(string_arg_number)
+    values = {
+        "port_name": port_name,
+        "short_description" : short_description,
+        "dir_name" : dir_name,
+        "arg_number" : arg_number,
+    }
+
+    #Fill in blank values with defaults
+    for key in values:
+        if values[key] == "":
+            values[key] = defaults[key]
+    return values
+
+def make_namespace(deployment, cwd):
+    # Form the namespace from the path to the deployment
+    namespace_path = cwd.relative_to(deployment)
+    deployment_list = str(deployment).split("/")
+    deployment_dir = deployment_list[-1]
+    whole_path = deployment_dir + "/" + str(namespace_path)
```

Comment:
One totally optional tip is it might be worth considering using python's [pathlib](https://docs.python.org/3/library/pathlib.html) for handling path manipulation like this. Whenever I see usage of raw `/` symbols I get a little nervous about they perform under certain edge cases, like with `a//b/c`. Pathlib is nice because it helps handle paths in an OS independent way.

Reply:
Ok, I changed this function to use pathlib and now it should work independent of the OS!